keep-ui/app/settings/add-user-modal.tsx

```diff
@@ -48,190 +43,163 @@ const roleOptions: RoleOption[] = [
+      try {
+        const response = await fetch(`${getApiURL()}/settings/users`, {
+          method: "POST",
+          headers: {
+            Authorization: `Bearer ${accessToken}`,
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(data),
+        });
 
-    // Validate form
-    if (!isFormValid()) {
-      setAddUserError("Please fill out all fields");
-      return;
-    }
-    // Validate selected role
-    if (!selectedRole) {
-      setAddUserError("Please select a role");
-      return;
-    }
+        if (response.ok) {
+          const newUser = await response.json();
+          setUsers(prevUsers => [...prevUsers, newUser]);
+          handleClose();
+        } else {
+          const errorData = await response.json();
+          // if 'detail' in errorData:
+          if (errorData.detail) {
+            setError('apiError', { type: 'manual', message: errorData.detail });
+          }
+          else{
+            setError('apiError', { type: 'manual', message: errorData.message || 'Failed to add user' });
+          }
 
-    // Make the API call to add the user
-    const response = await fetch(`${getApiURL()}/settings/users`, {
-      method: "POST",
-      headers: {
-        Authorization: `Bearer ${accessToken}`,
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({ email, role: selectedRole, password }),
-    });
+        }
+      } catch (error) {
+        setError('apiError', { type: 'manual', message: 'An unexpected error occurred' });
+      }
+    };
 
-    if (response.ok) {
-      const newUser = await response.json();
-      setUsers((currentUsers) => [...currentUsers, newUser]);
-      // Reset form and close modal on successful addition
-      handleOnClose();
-    } else {
-      const errorData = await response.json();
-      setAddUserError(
-        errorData.message ||
-          errorData.detail ||
-          "Failed to add user (unknown error)"
-      );
-    }
-  };
+    const handleClose = () => {
+        clearErrors('apiError');
+        reset();
+        onClose();
+      };
 
-  const handleOnClose = () => {
-    onClose();
-    setEmail("");
-    setSelectedRole("");
-    setPassword("");
-    setAddUserError("");
-  };
-
-  const handleEmailChange = (email: string) => {
-    setEmail(email);
-    if (validateEmail(email)) {
-      setEmailError("");
-    } else {
-      setEmailError("Please enter a valid email address.");
-    }
-  };
+    return (
+      <Dialog as="div" className="fixed inset-0 z-10 overflow-y-auto" open={isOpen} onClose={handleClose}>
+        <div className="flex items-center justify-center min-h-screen">
+          <Dialog.Panel className="bg-white p-4 rounded" style={{ width: "400px", maxWidth: "90%" }}>
+            <Dialog.Title>Add User</Dialog.Title>
+            <form
+                onSubmit={e => {
+                    clearErrors()
+                    handleSubmit(onSubmit)(e)
```

Comment:
If you get rid of the `clearErrors`, then you can [simply pass](https://react-hook-form.com/docs/useform/handlesubmit) `handleSubmit` to `onSubmit` like so:

```tsx
<form onSubmit={handleSubmit(onSubmit)}>
  {/* inputs here */}
</form>
```

Reply:
yea that was the situation in the beginning